### ISSUE TITLE
Dead patients can fail to die when expected

### ISSUE THREAD

**Issue by ledowong** [CorsixTH/CorsixTH#666]
I finish Campaign mode, and now i try my best to crash the game, and i did it =)

save:
https://drive.google.com/file/d/0B05D0Cat5usfRlBGWXp5ck5yYk0/view?usp=sharing

video:
http://youtu.be/ssn5DT8MRmM

1) at Sept 24, the scanner stop...... no error but just stop..
2) ignore issue 1, keep the game running, it will keep popup errors, trace below.

```
A stack trace is included below, and the handler has been disconnected.
C:\Program Files\CorsixTH\Lua\humanoid_actions\walk.lua:227: attempt to index field 'queue' (a nil value)
stack traceback:
        C:\Program Files\CorsixTH\Lua\humanoid_actions\walk.lua: in function 'action_walk_raw'
        C:\Program Files\CorsixTH\Lua\humanoid_actions\walk.lua:187: in function
 'timer_function'
        C:\Program Files\CorsixTH\Lua\entity.lua:215: in function 'tick'
        C:\Program Files\CorsixTH\Lua\world.lua:1103: in function 'onTick'
        C:\Program Files\CorsixTH\Lua\app.lua:852: in function 'dispatch'
        C:\Program Files\CorsixTH\Lua\app.lua:753: in function <C:\Program Files\CorsixTH\Lua\app.lua:748>

Warning: No event handler for timer
```

**Comment by SimplyRyan** [comment]
#666 &  #682 look to be the same issue.

**Comment by mugmuggy** [comment]
The first error the scanner stopping is a result of a dead (health = 0) patient entering the room, and gets a similar issue to a few other issues (still open). The patient in this instance is still in the queue (has idle, queue, walk) into room, where the die action should interrupt that walk but it hasn't for whatever reason - normally from a 'must happen' walk that occurs with that queues up the idle/queue/walk action again, living the die dangling on the end to be truncated. Scanner and decontamination both have issues with dead patients specifically, the goal would be to have dead patients die in all cases.

The next error is a dying patient that is attempting to access a tile in the room to die, he meanders to a position where the fall direction is influenced by patients/objects on the surrounding tiles and thus attempts to move y-1 which is within the room. This has been addressed since 0.4.

**Comment by lewri** [comment]
I don't think this one should happen anymore (the history of #2310 's original PR likely solved this). Infact, the first error by Mugmuggy's description is what #2310 (and original PR ) fixed.
Closing now.
